migrations/migration_5_share_gob_to_ssz_test.go

```
@@ -0,0 +1,83 @@
+		shareGOBHex, err := hex.DecodeString(shareGOBHexString)
+		assert.NoError(t, err)
+		shareGOBGot := &storageShareGOB{}
+		err = shareGOBGot.Decode(shareGOBHex)
+		assert.NoError(t, err)
```

do we have any conventions regarding when to use `assert` versus `require` ? :) they are quite similar in its behavior. Frankly, I used to use `require` (considering AAA) in Arrange step and `assert` in Assert step :) From what I saw in the tests we mainly use only `require` everywhere. It does not really matter much, although maybe at some point we could align on that

---

I don't think there is a strict set of guidelines we use, 

but I myself try to use `require` whenever I can just to "fail fast" - since it reduces noise in output and in 99.9% cases the error you are looking for will be the first one

for all other cases where seeing just the very first error isn't enough I would use `assert`,

here I just mistakenly used `assert` - fixed in https://github.com/ssvlabs/ssv/pull/2002/commits/ba4b807e8ac76e3bea3d3dfcabc8f2b6f99ebb6f now

---

thank you for explaining. No, it's good, I do not have any preferences nor think it has any importance of making it conventional, especially considering they behave differently. So probably, depending on the case we can use either `require` or `assert`